clients/common/nm-vpn-helpers.c

```diff
@@ -247,3 +253,522 @@ nm_vpn_openconnect_authenticate_helper (const char *host,
+		 *
+		 * Note that wg-quick doesn't do that.
+		 *
+		 * Neither `wg setconf` nor `wg-quick` does a strict parsing.
+		 * We don't either. Just try to interpret the file (mostly) the same as
+		 * they would.
+		 */
+		{
+			gsize l, n;
+
+			n = 0;
+			for (l = 0; (ch = line[l]); l++) {
+				if (g_ascii_isspace (ch)) {
+					/* wg-setconf strips all whitespace before parsing the content. That means,
+					 * *[I nterface]" will be accepted. We do that too. */
+					continue;
+				}
+				if (ch == '#')
+					break;
+				line[n++] = line[l];
+			}
+			if (n == 0)
+				continue;
+			line[n] = '\0';
+		}
+
+		if (g_ascii_strcasecmp (line, "[Interface]") == 0) {
+			if (!_wg_complete_peer (&data_peers,
+			                        g_steal_pointer (&current_peer),
+			                        current_peer_start_line_nr,
+			                        filename,
+			                        error))
+				return FALSE;
+			line_context = LINE_CONTEXT_INTERFACE;
+			continue;
+		}
+
+		if (g_ascii_strcasecmp (line, "[Peer]") == 0) {
+			if (!_wg_complete_peer (&data_peers,
+			                        g_steal_pointer (&current_peer),
+			                        current_peer_start_line_nr,
+			                        filename,
+			                        error))
+				return FALSE;
+			current_peer_start_line_nr = line_nr;
+			current_peer = nm_wireguard_peer_new ();
+			line_context = LINE_CONTEXT_PEER;
+			continue;
+		}
+
+		if (line_context == LINE_CONTEXT_INTERFACE) {
+
+			if (line_match (line, "Address", &matched_key, &value)) {
+				GPtrArray **p_data_addr;
+				NMIPAddr addr_bin;
+				int addr_family;
+				int prefix_len;
+
+				if (!nm_utils_parse_inaddr_prefix_bin (AF_UNSPEC,
+				                                       value,
+				                                       &addr_family,
+				                                       &addr_bin,
+				                                       &prefix_len))
+					goto fail_invalid_value;
+
+				p_data_addr =   (addr_family == AF_INET)
+				              ? &data_addr_v4
+				              : &data_addr_v6;
+
+				if (!*p_data_addr)
+					*p_data_addr = g_ptr_array_new_with_free_func ((GDestroyNotify) nm_ip_address_unref);
+
+				g_ptr_array_add (*p_data_addr,
+				                 nm_ip_address_new_binary (addr_family,
+				                                           &addr_bin,
+				                                             prefix_len == -1
+				                                           ? ((addr_family == AF_INET) ? 32 : 128)
+				                                           : prefix_len,
+				                                           NULL));
+				continue;
+			}
+
+			if (line_match (line, "MTU", &matched_key, &value)) {
+				i64 = _nm_utils_ascii_str_to_int64 (value, 0, 0, G_MAXUINT32, -1);
+				if (i64 == -1)
+					goto fail_invalid_value;
+
+				/* TODO: not yet implemented. Ignore the value.
+				 *
+				 * wg-quick accepts the "MTU" value, but it also fetches routes to
+				 * autodetect it. */
```

Comment:
NetworkManager is in a uniquely better position to manage this than wg-quick. Namely, it knows about and monitors all of the links already. Using the existing link monitoring, resolve the endpoint's interface, subtract 80, and keep that set.

Reply:
I don't think that link-monitoring suffices, because it's non-obvious which is the "entpoint's interface". It would require doing what`wg-quick` does, which looks up the interface (or the MTU) via `ip route get`. Basically, on every tiny change of IP configuration, NetworkManager would need to call `ip route get` for each endpoint, which seems also to have issues to scale. 

On #305 the MTU will simply be configurable, but in absence of an explicit value, it's left unset. NetworkManager also does that for other "ip-tunnel" types, where the same issue exists, and if this would be improved, it should be improved for other tunnel types as well.

Do you know how does systemd-networkd handles this?

Reply:
> Basically, on every tiny change of IP configuration, NetworkManager would need to call ip route get for each endpoint, which seems also to have issues to scale.

No, the routing table is supposed to be extremely fast, since it's queried for incoming packets. IIRC, a cache was recently removed, even, because the main routing table implementation was considered sufficiently fast.

Reply:
With #305 merged, I think what we want to do about MTU is now implemented. The import-code of this branch is also adjusted.

We may improve futher, but not as part of this pull-request. We should open a separate issue, if there are concrete suggestions what to do exactly.